### DISCUSSION TITLE
YouTube (Private) Video Download Failed

### DISCUSSION THREAD

**Discussion by debmalya001** [aclap-dev/video-downloadhelper#720] [Help and Q&A]
### Your question

{
  "version": "9.0.1.13",
  "target": "mozilla",
  "channel": "stable",
  "page_url": "https://www.youtube.com/watch?v=tsrSWHCjlMQ",
  "error": {
    "error": "coapp_failure",
    "details": "SideDownload error: : [in#1 @ 000001f254632e80] Error opening input: Server returned 403 Forbidden (access denied)\r\nError opening input file https://rr4---sn-cvh7knzd.c.youtube.com/videoplayback?expire=1716859709&ei=3d5UZrb_JPiljuMPs--_oAQ&ip=45.112.240.1&cp=X19QZ2d2d2gtOE1ETUtXSkk6Y1dfSHByQkJ0bVo5UmU0UkhiTzFXN2NDTVFiZDBkT0l5ZjFORzRGR1hFUg&id=o-ALhBbuGctg9rOMHv2n8s6DJjUBMn8m5UtwHZFzJvqDHZ&itag=136&aitags=134%2C136%2C160&source=youtube&requiressl=yes&xpc=EgVo2aDSNQ%3D%3D&mh=jL&mm=32%2C26&mn=sn-cvh7knzd%2Csn-h5576n7r&ms=su%2Conr&mv=m&mvi=4&pl=24&sc=yes&ctier=A&pfa=5&initcwndbps=1046250&hightc=yes&siu=1&bui=AWRWj2SoJXfTlzpqRFcDB9TBP-EUz-GDLpfMUFcY1Czp5CROolgFdWhj1XZTHKO1V_wq_OuwuQ&spc=UWF9f04w8Qu70vtiU-kXrHlzAwXDMvNc7CSxYCTuGNLaHKLUA3wYKXg66OBbYuMuK_XGgSXPHg&vprv=1&prv=1&svpuc=1&mime=video%2Fmp4&ns=dpUTyf1D53DEU-FNdwY7iZAQ&rqh=1&gir=yes&clen=57707857&dur=479.045&lmt=1709694823247005&mt=1716837647&fvip=2&keepalive=yes&c=WEB&sefc=1&txp=6219224&n=QjtPFKC87JgpMw&sparams=expire%2Cei%2Cip%2Ccp%2Cid%2Caitags%2Csource%2Crequiressl%2Cxpc%2Cctier%2Cpfa%2Chightc%2Csiu%2Cbui%2Cspc%2Cvprv%2Cprv%2Csvpuc%2Cmime%2Cns%2Crqh%2Cgir%2Cclen%2Cdur%2Clmt&sig=AJfQdSswRAIgTb8gPViZlHIa9J3MIKaq5oAInfU_nO1tlQGvn3O6jBACIBVGKllGBCDkgt-8jkIbwgDmQKAklmGJjE0BIu19Q5lm&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Csc%2Cinitcwndbps&lsig=AHWaYeowRQIhAOPbdbmAFvoiD8xYCzg1fvUn92xSqJs2Yal2SOzFUcnKAiBs33Y1YEdHkyKcpgAFAEGXumM1KJuyK7-Hp1GV8XjVUg%3D%3D.\r\nError opening input files: Server returned 403 Forbidden (access denied)\r\n",
    "id": "coapp_failure_45f27300-ecb8-40b7-a07f-6cfd8d11b221",
    "downloadable_id": "downloadable_1641463929225166"
  }
}

### Addon & CoApp version and details

version: 9.0.1.13
target: mozilla
channel: stable
lang: en-US
coapp: {"found":true,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"unset":true}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:126.0) Gecko/20100101 Firefox/126.0
{
    "used_sidebar_button": true,
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "use_sidebar": true,
    "open_count_store": 8
}

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

There have been other reports of similar problems.  Both YouTube private videos like yours & generally accessible but restricted YouTube videos that require a login seem to pose problems for VDH.  Also, it seems that anybody with a paid subscription on YouTube who is logged in has trouble getting VDH to work.  Logging out clears the problem.  I have a plain old free Google account  & VDH works for me on the rare occasion I visit YouTube while I'm logged in.  This is a problem that has been around for a few years.  I have no idea when, if ever, this will be addressed.

It is possible your content might be protected by DRM.  That would be indicated by a small chain-link icon in the top of the Firefox window, to the left of the URL.  If you see that, the content is protected by DRM & VDH will not download it.
